--- gst-plugins-good1.0-1.20.3/gst-libs/gst/basecamerabinsrc/meson.build	2023-01-23 14:02:49.996555705 +0100
+++ gst-plugins-good1.0-1.20.3/gst-libs/gst/basecamerabinsrc/meson.build	2023-01-23 14:02:54.588565551 +0100
@@ -25,6 +25,7 @@
 library_def = {'lib': gstbasecamerabin}
 pkg_name = 'gstreamer-bad-base-camerabinsrc-1.0'
 gen_sources = []
+static_build = get_option('default_library') == 'static'
 if build_gir and not static_build
   gir = {
     'sources' : camerabin_sources + camerabin_headers,
